pandas/tests/plotting/test_frame.py

```diff
@@ -1682,6 +1682,21 @@ def test_hist_df(self):
         axes = df.plot.hist(rot=50, fontsize=8, orientation="horizontal")
         self._check_ticks_props(axes, xrot=0, yrot=50, ylabelsize=8)
 
+    @pytest.mark.parametrize(
+        "weights", [0.1 * np.ones(shape=(100,)), 0.1 * np.ones(shape=(100, 2))]
+    )
+    def test_hist_weights(self, weights):
+        # GH 33173
+        np.random.seed(0)
+        df1 = pd.DataFrame(dict(zip(["A", "B"], np.random.randn(2, 100,))))
+
+        ax1 = _check_plot_works(df1.plot, kind="hist", weights=weights)
+
+        df2 = pd.DataFrame(dict(zip(["A", "B"], 0.1 * np.random.randn(2, 100,))))
+        ax2 = _check_plot_works(df2.plot, kind="hist")
+
+        self._check_data(ax1, ax2)
```

Comment:
Is there a way in the test to check that the weights are appropriately rendered by column?

Reply:
thanks for taking a look! @WillAyd 

good point, i slightly changed the test, and i think: in order to check weights to be rendered appropriately, we could use `height` in patches, if for instance, weights for all is `0.1` as in original example, the height of plot should be `0.1` times of original one without weights. 

what do you think of this approach?